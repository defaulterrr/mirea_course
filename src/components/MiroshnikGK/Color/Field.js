import React from 'react';
import PropTypes from 'prop-types';

const field = (props) => {
  const { value } = props;
  const { changed } = props;
  return (
    <div className="inputGK">
      <input type="text" value={value} onChange={changed} />
    </div>
  );
};

field.propTypes = {
  value: PropTypes.string,
  changed: PropTypes.func.isRequired,
};

field.defaultProps = {
  value: '',
};
export default field;
