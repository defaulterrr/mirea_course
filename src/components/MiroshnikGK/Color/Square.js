import React from 'react';
import Radium from 'radium';
import PropTypes from 'prop-types';

const square = (props) => {
  const { CMYK } = props;
  const { value } = props;
  const h4style = {
    height: '25px',
  };
  const rectKeyframes = Radium.keyframes({
    '0%': { transform: 'rotate(-3deg)' },
    '100%': { transform: 'rotate(3deg)' },
  });
  const rectanglestyle = {
    width: '300px',
    height: '300px',
    background: value,
    marginTop: '30px',
    border: '5px solid white',
    animation: '2s infinite alternate ease-in-out',
    animationName: rectKeyframes,
  };
  return (
    <div className="SquareGK">
      <div className="rectangleGK" style={rectanglestyle} />
      <h4 style={h4style}>
        CMYK:
        <br />
        {CMYK}
      </h4>
    </div>
  );
};

square.propTypes = {
  CMYK: PropTypes.string,
  value: PropTypes.string,
};

square.defaultProps = {
  CMYK: '',
  value: '',
};

export default Radium(square);
