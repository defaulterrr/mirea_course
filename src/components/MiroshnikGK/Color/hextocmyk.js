function hexToCMYK(color) {
  let computedC = 0;
  let computedM = 0;
  let computedY = 0;
  let computedK = 0;

  if (typeof color !== 'string') {
    return undefined;
  }

  if (color.charAt(0) !== '#') {
    return '';
  }
  const hex = (color.charAt(0) === '#') ? color.substring(1, 7) : color;

  if (hex.length !== 6) {
    return '';
  }
  if (/[0-9a-f]{6}/i.test(hex) !== true) {
    return '';
  }

  const r = parseInt(hex.substring(0, 2), 16);
  const g = parseInt(hex.substring(2, 4), 16);
  const b = parseInt(hex.substring(4, 6), 16);

  // BLACK
  if (r === 0 && g === 0 && b === 0) {
    computedK = 1;
    return '0 0 0 1';
  }

  computedC = 1 - (r / 255);
  computedM = 1 - (g / 255);
  computedY = 1 - (b / 255);

  const minCMY = Math.min(computedC, Math.min(computedM, computedY));

  computedC = (computedC - minCMY) / (1 - minCMY);
  computedC = computedC.toString().slice(0, 5);
  computedM = (computedM - minCMY) / (1 - minCMY);
  computedM = computedM.toString().slice(0, 5);
  computedY = (computedY - minCMY) / (1 - minCMY);
  computedY = computedY.toString().slice(0, 5);
  computedK = minCMY;
  computedK = computedK.toString().slice(0, 5);

  return (`${computedC} ${computedM} ${computedY} ${computedK}`);
}

export default hexToCMYK;
