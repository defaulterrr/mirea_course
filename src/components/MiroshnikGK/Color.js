import React, { Component } from 'react';
import Field from './Color/Field';
import Square from './Color/Square';
import hexToCMYK from './Color/hextocmyk';
import { StyleRoot } from 'radium'

class Color extends Component {

  state = {
    value: '',
    CMYK: '',
  };

  handleValueChange = (event) => {
    this.setState(
      {
        value: event.target.value,
        CMYK: hexToCMYK(event.target.value),
      }
    );
  }

  render() {
    return (
      <StyleRoot>
        <div className="Color">
          <Field id="field" value={this.state.value} changed={this.handleValueChange} />
          <Square id="square" value={this.state.value} CMYK={this.state.CMYK} />
        </div>
      </StyleRoot>
    );
  }
}
export default Color;
