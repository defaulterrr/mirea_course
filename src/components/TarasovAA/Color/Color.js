import React, { Component } from 'react';
import { StyleRoot } from 'radium';
import Field from './Field';
import Square from './Square';
import hexToCMYK from './hextocmyk';

class Color extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      CMYK: '',
    };
  }

  handleValueChange = (event) => {
    this.setState(
      {
        value: event.target.value,
        CMYK: hexToCMYK(event.target.value),
      },
    );
  }

  render() {
    const { value } = this.state;
    const { CMYK } = this.state;
    return (
      <StyleRoot>
        <div className="Color">
          <Field id="field" value={value} changed={this.handleValueChange} />
          <Square id="square" value={value} CMYK={CMYK} />
        </div>
      </StyleRoot>
    );
  }
}

export default Color;
