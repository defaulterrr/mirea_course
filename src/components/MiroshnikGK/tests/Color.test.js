import React from 'react';
import { create } from 'react-test-renderer';
import Color from '../Color';

describe('[RENDER] Color â€“ Snapshot test', () => {
  test('Matches the snapshot', () => {
    const color = create(<Color />);
    expect(color.toJSON()).toMatchSnapshot();
  });
});
