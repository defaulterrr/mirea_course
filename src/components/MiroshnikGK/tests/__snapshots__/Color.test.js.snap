// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Color â€“ Snapshot test Matches the snapshot 1`] = `
<div
  data-radium={true}
>
  <div
    className="Color"
    data-radium={true}
  >
    <div
      className="inputGK"
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="SquareGK"
      data-radium={true}
    >
      <div
        className="rectangleGK"
        data-radium={true}
        style={
          Object {
            "animation": "2s infinite alternate ease-in-out",
            "animationName": "radium-animation-baec9553",
            "background": "",
            "border": "5px solid white",
            "height": "300px",
            "marginTop": "30px",
            "width": "300px",
          }
        }
      />
      <h4
        data-radium={true}
        style={
          Object {
            "height": "25px",
          }
        }
      >
        CMYK:
        <br />
        
      </h4>
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@keyframes radium-animation-baec9553 {
0%{transform: rotate(-3deg);}
100%{transform: rotate(3deg);}
}
",
      }
    }
  />
</div>
`;
